lib/pry/input_completer.rb

```diff
@@ -239,4 +227,39 @@ def build_path(input)
     end
     return path, input
   end
+
+  private
+
+  def self.fetch_obj_space_collection
```

Comment:
@jazzonmymind I'm almost ready with similar to this PR refactoring, I'll make PR soon. I think method cache should be global, because it doesn't depend on pry's state.